driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -143,11 +167,17 @@ public void remove() {
             }
             wrapped.close();
 
+            changeStreamOperation.startAfter(null);
             if (resumeToken != null) {
-                changeStreamOperation.startAfter(null);
                 changeStreamOperation.startAtOperationTime(null);
-                changeStreamOperation.startOperationTimeForResume(null);
                 changeStreamOperation.resumeAfter(resumeToken);
+            } else if (changeStreamOperation.getStartAtOperationTime() != null
```

Comment:
I still need to take a closer look at this logic.  @DmitryLukyanov can you look too?